### PR TITLE
ENT-7002: Added OpenTelemetry content

### PR DESCRIPTION
**Pull request by amiegrace12**
Preview built here: https://amie.preview.docs.r3.com/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.html and https://amie.preview.docs.r3.com/en/platform/corda/4.10/enterprise/node/setup/corda-configuration-fields.html

### CODE REVIEW

**Review by yarivus — APPROVED** [review]
LGTM

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+## Client API
+
+OpenTelemetry span generation is also incorporated into the RPC client API. If you want your Corda client code to generate spans, you need to include the OpenTelemetry SDK as a dependency. The methods described above only apply to the Corda node. 
+
+<!-- (TODO: Give example of dependencies needed). -->
+
+## Creating your own Spans
+
+The OpenTelemetry API may be used in your flows and in your client code to create spans and baggage. To get an instance of the OpenTelemetry API in a flow, make the following call:
+`val openTelemetry: OpenTelemetry? = serviceHub.telemetryService.getTelemetryHandle(OpenTelemetry::class.java)`.
```
Comment (adelel1) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:68]: The line starting val is code. So should be in its own code box (diff font?).
Currently getting munged with existing text.

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+From the client API, where RPC is a CordaRPCConnection, you would use the following:
+`val openTelemetry: OpenTelemetry? = rpc.getTelemetryHandle(OpenTelemetry::class.java)`.
+
+When creating your own spans, you can also create your own baggage. If you create your own baggage, it will also be sent to other nodes, and you can specify if you want this baggage to be copied to span tags. If you do, all of the spans involved in the transaction for that node will also get a copy of the baggage. This can be enabled with the following parameter:
+`telementry.copyBaggageToTags = true`.
+
+The default value of this setting is `false`.
+
+{{< note >}}
+Within a checkpoint, only the span ID has a checkpoint, meaning spans do not survive a node restart. If the node restarts, the parent span information will be lost, and new spans will be generated for the flows. The root span after the node restart won't know who the parent span was before the node restart.
```
Comment (adelel1) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:79]: Slight change to first sentence...

When checkpointing only the span id is checkpointed, meaning spans do not...

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+The current implementation of OpenTelemetry will send spans to the backend when the flow or operation is completed. This is handled via the OpenTelemetry SDK. If the flow somehow gets stuck or does not complete, the span representing that flow will never reach the backend. It can be difficult to figure out what went wrong by just looking at the spans, as you will only see the complete spans.
+
+{{< note >}}
+If a child span doesn't complete, parent spans also do not complete.
+{{< /note >}}
+
+As an alternative, the Corda OpenTelemetry component sends a start or end span to the backend when a flow or operation starts or stops, in addition to the normal spans sent for the operation. This is effectively a start flow span event and an end flow span event. With this view of the spans, it becomes easier to determine where the flow got stuck, as it will be the lowest child without an end span event. 
+
+{{< note >}}
+These start and end span events are only generated for spans that Corda generated. If you create a span in your own flow code, you won’t see equivalent start and end span events for your flows, as Corda knows nothing of them before the node restart.
```
Comment (adelel1) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:93]: Can you delete "before the node restart" at the end of this sentence.

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+<!-- (TODO: Give example of dependencies needed). -->
+
+## Creating your own Spans
+
+The OpenTelemetry API may be used in your flows and in your client code to create spans and baggage. To get an instance of the OpenTelemetry API in a flow, make the following call:
+`val openTelemetry: OpenTelemetry? = serviceHub.telemetryService.getTelemetryHandle(OpenTelemetry::class.java)`.
+
+From the client API, where RPC is a CordaRPCConnection, you would use the following:
+`val openTelemetry: OpenTelemetry? = rpc.getTelemetryHandle(OpenTelemetry::class.java)`.
```
Comment (adelel1) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:71]: Same code comment here. Should be in its own code box or diff font etc.

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md (hunk 1 above)
Comment (amiegrace12) [reply content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md]: I added in a Kotlin code block, can you check if it looks ok? Thanks!

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md (hunk 2 above)
Comment (amiegrace12) [reply content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md]: Done

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md (hunk 3 above)
Comment (amiegrace12) [reply content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md]: Done

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+Corda OpenTelemetry can also be enabled or disabled via a node configuration parameter, such as `telementry.openTelemetryEnabled = true`.
+
+If this setting is true and the OpenTelemetry SDK has been linked to Corda as described above, the node will generate spans. If this setting is false, the node won't generate spans, even if the node is linked with the SDK.
+
+## Client API
+
+OpenTelemetry span generation is also incorporated into the RPC client API. If you want your Corda client code to generate spans, you need to include the OpenTelemetry SDK as a dependency. The methods described above only apply to the Corda node. 
+
+<!-- (TODO: Give example of dependencies needed). -->
```
Comment (adelel1) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:63]: This answers the TODO:

You can link to the opentelemetry SDK from a client project in two ways. By compiling against the open telemetry driver which includes the open telemetry SDK. Or by running the client with the open telemetry java agent attached.

To use the open telemetry driver add the following dependency to the build.gradle file of the client.

dependencies { 
...
compile "net.corda:corda-opentelemetry-driver:$corda_release_version"
}

Alternatively you can attach your client to the open telemetry java agent via the following command line. 

java -javaagent:%PATH-TO-OT-JAVAAGENT%/opentelemetry-javaagent.jar -Dotel.service.name=YOUR-SERVICE-NAME -jar ./your-client.jar

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    }
+``` 
+
+From the client API, where RPC is a CordaRPCConnection, you would use the following:
+
+```kotlin
+val openTelemetry: OpenTelemetry? = rpc.getTelemetryHandle(OpenTelemetry::class.java)
+    }
+``` 
+
```
Comment (adelel1) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:80]: Once you have the open telemetry handle then you may call the open telemetry api as documented in the open telemetry documentation.

File: content/en/platform/corda/4.10/community/corda-configuration-fields.md [hunk 7]
```diff
@@ -511,6 +511,32 @@ Deprecated. Use rpcSettings instead.**
 
   *Default:* not defined
 
+## `telemetry`
+
+New configuration fields for telemetry. See the [OpenTelemetry](../../../enterprise/node/operating/monitoring-and-logging/opentelemetry.md) section for more information.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/corda-configuration-fields.md:516]:
```suggestion
There are new configuration fields for telemetry. See the [OpenTelemetry](../../../enterprise/node/operating/monitoring-and-logging/opentelemetry.md) section for more information. 
```

File: content/en/platform/corda/4.10/community/corda-configuration-fields.md [hunk 8]
```diff
@@ -511,6 +511,32 @@ Deprecated. Use rpcSettings instead.**
...
+  * *Default:* true
+* `spanStartEndEventsEnabled`
+  * When Corda generates spans for flows and certain significant operations, it has the capability to generate a span for starting the operation, ending the operation, and generating a single span to cover the whole operation. This can be useful to determine where a flow is stuck, as you will only see the start spans, and not the end spans. This is not standard OpenTelemetry behaviour, and it could also result in a lot of spans flooding the network. Setting this field to `true` will enable it. 
+  * *Default:* false
+* `copyBaggageToTags`
+  * If set to `true`, this parameter will cause baggage to be copied to tags when generating spans. Baggage are fields which can be passed around with the invocation of OpenTelemetry.
+  * *Default:* false
+* `simpleLogTelemetryEnabled`
+  * Enables an alternative form of telemetry. If this field is set to `true`, log lines are written to logs when a flow or significant operation is started, or ended. The log line specifies a trace ID, which allows flows to be matched up across nodes.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/corda-configuration-fields.md:528]:
```suggestion
  * Enables an alternative form of telemetry. If this field is set to `true`, log lines are written to logs when a flow or significant operation is started or ended. The log line specifies a trace ID, which allows flows to be matched up across nodes.
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+  corda-community-4-10:
+    identifier: corda-community-4-10-open-telemetry
+    parent: corda-community-4-10-corda-nodes-index
+    weight: 1250
+title: OpenTelemetry
+---
+
+# OpenTelemetry
+
+This section describes how to setup OpenTelemetry and the simple log telemetry.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:13]:
```suggestion
This topic describes how to set up OpenTelemetry and the simple log telemetry.
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+{{< /note >}}
+
+
+## Corda OpenTelemetry integration
+
+The Corda OpenTelemetry component links at compile time with the OpenTelemetry API. In order to produce spans, there is a runtime dependency on the OpenTelemetry SDK. 
+
+There are two ways to make the OpenTelemetry SDK available to the Corda node:
+
+* Include the `corda-opentelemetry-driver` into the drivers directory of the Corda node.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:31]:
```suggestion
* Add the file `corda-opentelemetry-driver-%VERSION%.jar` to the `drivers` directory of the Corda node.
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+## Using the OpenTelemetry driver
+
+To use the OpenTelemetry driver, copy `corda-opentelemetry-driver-%VERSION%.jar` into the drivers directory of the Corda node. This driver can be downloaded from Artifactory and is included in the release pack.
+
+## Using the OpenTelemetry Java agent
+
+To use the OpenTelemetry Java agent:
+
+1. Download the agent from [GitHub](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases). The file you need is `opentelemetry-javaagent.jar`.
+2. Start the Corda node with the following example command line: `java -Dcapsule.jvm.args=“-javaagent:/PATH-TO-OT-JAVAAGENT/opentelemetry-javaagent.jar -Dotel.service.name=YOUR-SERVICE-NAME” -jar corda.jar`.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:43]:
```suggestion
2. Start the Corda node with the following example command line:
   `java -Dcapsule.jvm.args=“-javaagent:/PATH-TO-OT-JAVAAGENT/opentelemetry-javaagent.jar -Dotel.service.name=YOUR-SERVICE-NAME” -jar corda.jar`.
   
   
   
   
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+3. Replace `PATH-TO-OT-JAVAAGENT` with the full path to where you placed the OpenTelemetry Java agent.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:44]:
```suggestion
3. Replace `PATH-TO-OT-JAVAAGENT` with the full path of the directory where you placed the OpenTelemetry Java agent.
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+{{< /note >}}
+
+
+## Setting up OpenTelemetry
+
+Corda has been tested with the OpenTelemetry collector and a Jaeger backend. Refer to the [OpenTelemetry](https://opentelemetry.io/docs/collector/) and [Jaeger](https://www.jaegertracing.io/) websites for details on how to setup a collector and Jaeger backend.
+
+Corda OpenTelemetry can also be enabled or disabled via a node configuration parameter, such as `telementry.openTelemetryEnabled = true`.
+
+If this setting is true and the OpenTelemetry SDK has been linked to Corda as described above, the node will generate spans. If this setting is false, the node won't generate spans, even if the node is linked with the SDK.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:58]:
```suggestion
If this setting is true and the OpenTelemetry SDK has been linked to Corda as described above, the node will generate spans. If this setting is false, the node will not generate spans, even if the node is linked with the SDK.
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+## Client API
+
+OpenTelemetry span generation is also incorporated into the RPC client API. If you want your Corda client code to generate spans, you need to include the OpenTelemetry SDK as a dependency. The methods described above only apply to the Corda node. 
+
+You can link to the OpenTelemetry SDK from a client project in two ways:
+
+* By compiling against the OpenTelemetry driver which includes the OpenTelemetry SDK. 
+* By running the client with the OpenTelemetry Java agent attached.
+
+To use the OpenTelemetry driver, add the following dependency to the build.gradle file of the client:
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:69]:
```suggestion
To use the OpenTelemetry driver, add the following dependency to the `build.gradle` file of the client:
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+...
+compile "net.corda:corda-opentelemetry-driver:$corda_release_version"
+}
+``` 
+
+Alternatively, you can attach your client to the OpenTelemetry Java agent via the following command line:
+
+`java -javaagent:%PATH-TO-OT-JAVAAGENT%/opentelemetry-javaagent.jar -Dotel.service.name=YOUR-SERVICE-NAME -jar ./your-client.jar`.
+
+## Creating your own Spans
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:82]:
```suggestion
## Creating your own spans
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+
+### Baggage 
+
+When creating your own spans, you can also create your own baggage. If you create your own baggage, it will also be sent to other nodes, and you can specify if you want this baggage to be copied to span tags. If you do, all of the spans involved in the transaction for that node will also get a copy of the baggage. This can be enabled with the following parameter:
+`telementry.copyBaggageToTags = true`.
+
+The default value of this setting is `false`.
+
+{{< note >}}
+When checkpointing, only the span id is checkpointed, meaning spans do not survive a node restart. If the node restarts, the parent span information will be lost, and new spans will be generated for the flows. The root span after the node restart won't know who the parent span was before the node restart.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:107]:
```suggestion
When checkpointing, only the span ID is checkpointed, meaning spans do not survive a node restart. If the node restarts, the parent span information will be lost, and new spans will be generated for the flows. The root span after the node restart will not know who the parent span was before the node restart.
```

File: content/en/platform/corda/4.10/community/opentelemetry.md [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+{{< /note >}}
+
+## Start and End Spans
+
+The current implementation of OpenTelemetry will send spans to the backend when the flow or operation is completed. This is handled via the OpenTelemetry SDK. If the flow somehow gets stuck or does not complete, the span representing that flow will never reach the backend. It can be difficult to figure out what went wrong by just looking at the spans, as you will only see the complete spans.
+
+{{< note >}}
+If a child span doesn't complete, parent spans also do not complete.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/community/opentelemetry.md:115]:
```suggestion
If a child span does not complete, parent spans also do not complete.
```

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 18]
```diff
@@ -0,0 +1,130 @@
...
+{{< /note >}}
+
+
+## Corda OpenTelemetry integration
+
+The Corda OpenTelemetry component links at compile time with the OpenTelemetry API. In order to produce spans, there is a runtime dependency on the OpenTelemetry SDK. 
+
+There are two ways to make the OpenTelemetry SDK available to the Corda node:
+
+* Include the `corda-opentelemetry-driver` into the drivers directory of the Corda node.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:30]:
```suggestion
* Add the file `corda-opentelemetry-driver-%VERSION%.jar` to the `drivers` directory of the Corda node.
```

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 19]
```diff
@@ -0,0 +1,130 @@
...
+## Using the OpenTelemetry driver
+
+To use the OpenTelemetry driver, copy `corda-opentelemetry-driver-%VERSION%.jar` into the drivers directory of the Corda node. This driver can be downloaded from Artifactory and is included in the release pack.
+
+## Using the OpenTelemetry Java agent
+
+To use the OpenTelemetry Java agent:
+
+1. Download the agent from [GitHub](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases). The file you need is `opentelemetry-javaagent.jar`.
+2. Start the Corda node with the following example command line: `java -Dcapsule.jvm.args=“-javaagent:/PATH-TO-OT-JAVAAGENT/opentelemetry-javaagent.jar -Dotel.service.name=YOUR-SERVICE-NAME” -jar corda.jar`.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:42]:
```suggestion
2. Start the Corda node with the following example command line:
   `java -Dcapsule.jvm.args=“-javaagent:/PATH-TO-OT-JAVAAGENT/opentelemetry-javaagent.jar -Dotel.service.name=YOUR-SERVICE-NAME” -jar corda.jar`.
```

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 20]
```diff
@@ -0,0 +1,130 @@
...
+3. Replace `PATH-TO-OT-JAVAAGENT` with the full path to where you placed the OpenTelemetry Java agent.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:43]:
```suggestion
3. Replace `PATH-TO-OT-JAVAAGENT` with the full path of the directory where you placed the OpenTelemetry Java agent.
```

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 21]
```diff
@@ -0,0 +1,130 @@
...
+{{< /note >}}
+
+
+## Setting up OpenTelemetry
+
+Corda has been tested with the OpenTelemetry collector and a Jaeger backend. Refer to the [OpenTelemetry](https://opentelemetry.io/docs/collector/) and [Jaeger](https://www.jaegertracing.io/) websites for details on how to setup a collector and Jaeger backend.
+
+Corda OpenTelemetry can also be enabled or disabled via a node configuration parameter, such as `telementry.openTelemetryEnabled = true`.
+
+If this setting is true and the OpenTelemetry SDK has been linked to Corda as described above, the node will generate spans. If this setting is false, the node won't generate spans, even if the node is linked with the SDK.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:57]:
```suggestion
If this setting is true and the OpenTelemetry SDK has been linked to Corda as described above, the node will generate spans. If this setting is false, the node will not generate spans, even if the node is linked with the SDK.
```

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 22]
```diff
@@ -0,0 +1,130 @@
...
+## Client API
+
+OpenTelemetry span generation is also incorporated into the RPC client API. If you want your Corda client code to generate spans, you need to include the OpenTelemetry SDK as a dependency. The methods described above only apply to the Corda node. 
+
+You can link to the OpenTelemetry SDK from a client project in two ways:
+
+* By compiling against the OpenTelemetry driver which includes the OpenTelemetry SDK. 
+* By running the client with the OpenTelemetry Java agent attached.
+
+To use the OpenTelemetry driver, add the following dependency to the build.gradle file of the client:
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:68]:
```suggestion
To use the OpenTelemetry driver, add the following dependency to the `build.gradle` file of the client:
```

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 23]
```diff
@@ -0,0 +1,130 @@
...
+
+### Baggage 
+
+When creating your own spans, you can also create your own baggage. If you create your own baggage, it will also be sent to other nodes, and you can specify if you want this baggage to be copied to span tags. If you do, all of the spans involved in the transaction for that node will also get a copy of the baggage. This can be enabled with the following parameter:
+`telementry.copyBaggageToTags = true`.
+
+The default value of this setting is `false`.
+
+{{< note >}}
+When checkpointing, only the span id is checkpointed, meaning spans do not survive a node restart. If the node restarts, the parent span information will be lost, and new spans will be generated for the flows. The root span after the node restart won't know who the parent span was before the node restart.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:106]:
```suggestion
When checkpointing, only the span ID is checkpointed, meaning spans do not survive a node restart. If the node restarts, the parent span information will be lost, and new spans will be generated for the flows. The root span after the node restart will not know who the parent span was before the node restart.
```

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 24]
```diff
@@ -0,0 +1,130 @@
...
+{{< /note >}}
+
+## Start and End Spans
+
+The current implementation of OpenTelemetry will send spans to the backend when the flow or operation is completed. This is handled via the OpenTelemetry SDK. If the flow somehow gets stuck or does not complete, the span representing that flow will never reach the backend. It can be difficult to figure out what went wrong by just looking at the spans, as you will only see the complete spans.
+
+{{< note >}}
+If a child span doesn't complete, parent spans also do not complete.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:114]:
```suggestion
If a child span does not complete, parent spans also do not complete.
```

File: content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md [hunk 25]
```diff
@@ -0,0 +1,130 @@
...
+{{< /note >}}
+
+As an alternative, the Corda OpenTelemetry component sends a start or end span to the backend when a flow or operation starts or stops, in addition to the normal spans sent for the operation. This is effectively a start flow span event and an end flow span event. With this view of the spans, it becomes easier to determine where the flow got stuck, as it will be the lowest child without an end span event. 
+
+{{< note >}}
+These start and end span events are only generated for spans that Corda has generated. If you create a span in your own flow code, you won’t see equivalent start and end span events for your flows, as Corda knows nothing of them.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md:120]:
```suggestion
These start and end span events are only generated for spans that Corda has generated. If you create a span in your own flow code, you will not see equivalent start and end span events for your flows, as Corda knows nothing of them.
```

File: content/en/platform/corda/4.10/enterprise/node/setup/corda-configuration-fields.md [hunk 26]
```diff
@@ -889,6 +889,32 @@ Keys and values of the map should be strings. e.g. `systemProperties = { "visual
 
 *Default:* not defined
 
+## `telemetry`
+
+New configuration fields for telemetry. See the [OpenTelemetry](../../../enterprise/node/operating/monitoring-and-logging/opentelemetry.md) section for more information.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.10/enterprise/node/setup/corda-configuration-fields.md:894]:
```suggestion
There are new configuration fields for telemetry. See the [OpenTelemetry](../../../enterprise/node/operating/monitoring-and-logging/opentelemetry.md) section for more information. 
```

**Review by paulmoloneyr3 — APPROVED** [review]

**Pull request merged**